### DISCUSSION TITLE
Is there a way to see the compiled code that Vitest executes?

### DISCUSSION THREAD

**Discussion by rossng** [vitest-dev/vitest#2242] [Q&A] [answered]
I'm having problems running my code under Vitest (a `ReferenceError` caused by a TypeScript decorator which doesn't happen when running the application itself).

To debug this is would be very useful to see the code that Vitest is actually building for execution - a bit like Vite's `vite build` command. Is there a way to do this?

**Comment by sheremet-va** [comment, marked as answer]
You can see compiled code in `vitest --ui` on module graph tab, or you can also use environmental variable `VITE_NODE_DEBUG_DUMP` as described in https://github.com/vitest-dev/vitest/tree/main/packages/vite-node#debugging

**Reply by rossng** [reply]
The UI mode is awesome 😮  

Seems you can't copy+paste the code from the module graph view though - might make a PR to fix that at some point. `VITE_NODE_DEBUG_DUMP` worked perfectly though. Thanks!
